### PR TITLE
Signature support: ECDSA and Ed25519

### PR DESCRIPTION
**Pull request by tarcieri**
Adds a toplevel `signature` module with `ecdsa` and `ed25519` submodules.

### CODE REVIEW

File: tests/signature/ed25519.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        let sig = Signature::from_bytes(&tweaked_sig[..]).unwrap();
+        assert!(verify_key.verify(vector.msg, &sig).is_err());
+    }
+}
+
+/// Ed25519 test vectors (from RFC 8032, converted to Rust bytestring literals)
+const TEST_VECTORS: &[TestVector] = &[
+    TestVector {
+        sk: b"\x9D\x61\xB1\x9D\xEF\xFD\x5A\x60\xBA\x84\x4A\xF4\x92\xEC\x2C\xC4\x44\x49\xC5\x69\x7B\x32\x69\x19\x70\x3B\xAC\x03\x1C\xAE\x7F\x60",
+        pk: b"\xD7\x5A\x98\x01\x82\xB1\x0A\xB7\xD5\x4B\xFE\xD3\xC9\x64\x07\x3A\x0E\xE1\x72\xF3\xDA\xA6\x23\x25\xAF\x02\x1A\x68\xF7\x07\x51\x1A",
```
Comment (newpavlov) [inline tests/signature/ed25519.rs:44]: Use `hex-literal` to make these vectors more compact?

Comment (tarcieri) [reply tests/signature/ed25519.rs:44]: Actually it's a bit tricky, since they're differently-sized...

Comment (newpavlov) [reply tests/signature/ed25519.rs:44]: IIRC `&hex!("0123")[..]` should do the job.

Comment (tarcieri) [reply tests/signature/ed25519.rs:44]: Yeah that should be fine. I guess on 1.47.0+ we can do `.as_ref()`?

Comment (newpavlov) [reply tests/signature/ed25519.rs:44]: Probably, yes.

**Pull request merged**